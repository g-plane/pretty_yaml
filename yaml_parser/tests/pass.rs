use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
