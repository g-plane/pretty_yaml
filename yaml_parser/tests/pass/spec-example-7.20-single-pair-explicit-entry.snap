---
source: yaml_parser/tests/pass.rs
---
ROOT@0..21
  DOCUMENT@0..20
    FLOW@0..20
      FLOW_SEQ@0..20
        L_BRACKET@0..1 "["
        WHITESPACE@1..2 "\n"
        FLOW_SEQ_ENTRIES@2..19
          FLOW_SEQ_ENTRY@2..18
            FLOW_PAIR@2..18
              FLOW_MAP_KEY@2..12
                QUESTION_MARK@2..3 "?"
                WHITESPACE@3..4 " "
                FLOW@4..12
                  PLAIN_SCALAR@4..12 "foo\n bar"
              WHITESPACE@12..13 " "
              COLON@13..14 ":"
              WHITESPACE@14..15 " "
              FLOW_MAP_VALUE@15..18
                FLOW@15..18
                  PLAIN_SCALAR@15..18 "baz"
          WHITESPACE@18..19 "\n"
        R_BRACKET@19..20 "]"
  WHITESPACE@20..21 "\n"
